HW#2 – Eugenia Kurmeneva

Gina-K opened the pull request:
For this project I've used SCSS, Webpack and BEM.

For build this project on your machine [check readme.txt](https://github.com/Gina-K/hw-2-responsive/blob/master/readme.txt)

Gina-K committed:
init

HTML from HW1

Gina-K committed:
Add resources

imgs and reset.css

Gina-K committed:
Configure environment

Gina-K committed:
Add file-loader and load import images

Gina-K committed:
Add basic styles

Gina-K committed:
Add basic styles for the small version of the widget

Except of flexes

Gina-K committed:
Add reset styles as .scss

Gina-K committed:
Complete small (mobile) version of the widget

Gina-K committed:
Add styles for the medium widget

Except of the carousel -- it's another component

Gina-K committed:
Add styles for the large (origin)  widget

Except of the carousel -- it's another component

Gina-K committed:
Add 12-column grid

Gina-K committed:
Add carousel

Gina-K committed:
Add basic styles for the carousel

Gina-K committed:
Complete the carousel

Gina-K committed:
Fix date font-size

Change original Paris.psd font weight for better readability

Gina-K committed:
Fix box-sizing mixin

Gina-K committed:
Fix carousel resizing

Gina-K committed:
Add prefixes

Gina-K committed:
Refactor working directory

Move reusable mixins to the separate file
Add dist's files to Git
Rename carousel.scss to carousel-flex.scss

Gina-K committed:
Add background to the page

Gina-K committed:
Add prefixes to the page background style

Gina-K committed:
Set environment for the 2nd (css grid) version

Gina-K committed:
Add grid styles for the main info block

Gina-K committed:
Add prefixes for grid properties

Gina-K committed:
Fix main info block grid for IE11

It was painful

Gina-K committed:
Add to git images in the dist folder

Gina-K committed:
Remove unnecessary padding property

Gina-K committed:
Add grid styles for the carousel

Gina-K committed:
Fix nth-child error

Gina-K committed:
Refactor

Gina-K committed:
Delete reset.css

Gina-K committed:
Fix carousel items alignment on IE

Also add empty lines to .html and import mixins to carousel

Gina-K committed:
Fix tags

Change <aside> and <p> to <div> according to HW1 discussion

Gina-K committed:
Change Webpack mode to minify .css files

jeanny-driver commented on dist/index.html:
```diff
@@ -0,0 +1,81 @@
+
+  <link href="https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700;900&display=swap" rel="stylesheet">
+  <link rel='stylesheet' href='weather-widget-flex.css'>
+<!--  <link rel='stylesheet' href='weather-widget-css-grid.css'>-->
```
Do you need need this commented code?

Gina-K replied:
Yeap. I use two different styles files and only one .html. So I switch between flex and grid version via comment the links.

jeanny-driver commented on dist/index.html:
```diff
@@ -0,0 +1,81 @@
+            <div class="weather-widget__day-of-week weather-widget__day-of-week_size_s weather-widget__day-of-week_size_m">Friday</div>
+            <div class="weather-widget__date weather-widget__date_size_m">August, 23rd</div>
+          </div>
+          <div class="weather-widget__space weather-widget__space_36px weather-widget__space_26px"></div>
```
It's a bad practice to use value of css property right in class names.

Gina-K replied:
[Fixed](https://github.com/Gina-K/hw-2-responsive/commit/ed1f6f58ffe0d49e883350185ad448e7496678c3)

jeanny-driver commented on dist/index.html:
```diff
@@ -0,0 +1,81 @@
+            <div class="weather-widget__date weather-widget__date_size_m">August, 23rd</div>
+          </div>
+          <div class="weather-widget__space weather-widget__space_36px weather-widget__space_26px"></div>
+          <div class="weather-widget__weather weather-widget__weather_ie-s weather-widget__weather_ie-m">
```
`ie`? Is this style applied only for IE?

Gina-K replied:
Yes, they're for IE only. For these selectors I use the properties with spesial -ms prefix. You can find them in the file [weather-widget-grid.scss](https://github.com/Gina-K/hw-2-responsive/blob/master/src/scss/weather-widget-grid.scss) on the lines 76-79 and 158-161.

jeanny-driver commented on dist/index.html:
```diff
@@ -0,0 +1,81 @@
+          </div>
+        </div>
+
+        <div class="weather-widget__carousel weather-widget__carousel_display_none weather-widget__carousel_display_block">
```
Don't use css properties in classes names.

Gina-K replied:
[Fixed](https://github.com/Gina-K/hw-2-responsive/commit/cc163f62e06ba2e4b1320fbce2ed4fddf77d7339)

jeanny-driver commented on dist/index.html:
```diff
@@ -0,0 +1,81 @@
+        </div>
+
+        <div class="weather-widget__carousel weather-widget__carousel_display_none weather-widget__carousel_display_block">
+          <div class="carousel carousel_size_m carousel_size_l">
```
Same. `size-m`, `size-1` are not appropriate for names. Check for other cases.

Gina-K replied:
But why not? BEM uses this way, please check [Modifiers](https://ru.bem.info/methodology/css/#%D0%BC%D0%BE%D0%B4%D0%B8%D1%84%D0%B8%D0%BA%D0%B0%D1%82%D0%BE%D1%80%D1%8B), an example with button.

jeanny-driver commented on dist/weather-widget-css-grid.css:
```diff
@@ -0,0 +1,335 @@
+article, aside, canvas, details, embed,
+figure, figcaption, footer, header, hgroup,
+menu, nav, output, ruby, section, summary,
+time, mark, audio, video {
```
You don't use most of this selectors in `html`. Delete them, please, and check for other similar cases.

Gina-K replied:
[Fixed](https://github.com/Gina-K/hw-2-responsive/commit/2587b8688532df96fe56df16c48b91ace68c93e0)

jeanny-driver commented on dist/weather-widget-flex.css:
```diff
@@ -0,0 +1,306 @@
+blockquote:before, blockquote:after,
+q:before, q:after {
+  content: '';
+  content: none; }
```
Two contents?

Gina-K replied:
Yes. This needs to support the older browsers. You can find more information about this question here: [https://stackoverflow.com/questions/6803250/content-vs-content-none](https://stackoverflow.com/questions/6803250/content-vs-content-none)

jeanny-driver commented on dist/weather-widget-css-grid.css:
```diff
@@ -0,0 +1,335 @@
+  white-space: nowrap; }
+
+.weather-widget__space {
+  display: none; }
```
If it's not displayed, why do you need it?

Gina-K replied:
Just because I use only one .html file. This element needs for the flex version and unuseful for the grid one -- so I just hide it.

jeanny-driver replied:
So if I want to see flex version I shouldn\t hide it?
It's not really sutable if I should comment/uncomment some code or change styles to switch from one view to another.

Gina-K committed:
Change selectors' names for the space element

Replace weather-widget__space_36px with weather-widget__space_big,
and weather-widget__space_26px with weather-widget__space_small

Gina-K committed:
Rename selectors for the carousel

Gina-K committed:
Clear reset.scss

Gina-K commented:
Thank you for the review and please check my fixes. :smiley_cat:

jeanny-driver commented on dist/index.html:
```diff
@@ -0,0 +1,81 @@
+<!doctype html>
```
BTW, for such project i'ts better to add directory `.gitignore` and list there files which shouldn't be pushed and can be generated via webpack, for example. You can add there `dist`, `node_modules`, `package-lock.json`
Didn't noticed first time, my fault.

Gina-K replied:
[Fixed](https://github.com/Gina-K/hw-2-responsive/commit/ba4ac11e6c8fd83b91a6e6d007b0cbb617aa1437)

Gina-K committed:
Add images to src folder

Gina-K committed:
Fix links for the images

jeanny-driver commented on dist/index.html:
```diff
@@ -0,0 +1,81 @@
+<!doctype html>
+<html lang="en">
```
Please arrange the widget in the center of the screen (Chrome)
![image](https://user-images.githubusercontent.com/71435688/95722495-ae9dfe80-0c7c-11eb-9a33-4d46e293ba34.png)
In mobile devices there is no carousel
![image](https://user-images.githubusercontent.com/71435688/95722623-d8572580-0c7c-11eb-8421-111784661c6c.png)
This is look in FF:
![image](https://user-images.githubusercontent.com/71435688/95722674-ead15f00-0c7c-11eb-8349-16bbbcefddcc.png)
According conditions the widget should be responsive...

Gina-K replied:
Fixed: [added the carousel to the small widget](https://github.com/Gina-K/hw-2-responsive/commit/3d37495ea27b5a4f7d0be7c43a298a2c2074fb28) and [centered the widget](https://github.com/Gina-K/hw-2-responsive/commit/6009b06bc2949842532959c6fd5855db584d2629)

jeanny-driver replied:
Looks much more better! There is still small issue with mobile device (our lovely carousel )
![image](https://user-images.githubusercontent.com/71435688/95882717-b21bad80-0d82-11eb-8291-e0949157b501.png)
I'm going to approve PR, but you still can play with carousel on mobile 📱

Gina-K replied:
Great, thank you so much! ❤️  I have some idea about the carousel, please take a look ([last commit](https://github.com/Gina-K/hw-2-responsive/commit/40b14c7c91328aaa3c7f75486c220cf69ac48641))

Gina-K committed:
Change project structure

Now I use only one .css file (instead of two) and choose what
version build via index.js import

Gina-K committed:
Add the carousel to the small widget

Gina-K committed:
Center the widget

Gina-K committed:
Clear dist folder

Gina-K committed:
Create .gitignore

Gina-K committed:
Create readme.txt with project run instruction

Gina-K committed:
Add more responsive behavior to the carousel
